tools/downloader/topologies.py

```diff
@@ -0,0 +1,99 @@
+import os
+import argparse
+
+from pathlib import Path
+
+import common
+import downloader
+import converter
+
+class Topology:     
+    '''
+    {}
+    Model task type: {}
+
+    License: {}
+    '''
+
+    def __init__(self, model_name, *, precision=None, download_dir=None):
+        '''
+        Downloads target model. If the precision doesn't set, loads all the available models.
+        If there is already downloaded model - retrieves the model from cache.
+        Creates an object which stores information about the downloaded model.
+
+        Parameters
+        ----------
+            model_name
+                Target model name.
+            precision
+                Target model precision. By default is None.
+            download_dir
+                Model download directory.
+        '''
+        if not download_dir: 
+            download_dir = Path(Path('.').resolve() / 'models')
+        cache_dir = Path(Path(download_dir) / '.cache')
+
+        parser = argparse.ArgumentParser()
+        args = argparse.Namespace(all=False, list=None, name=model_name, print_all=False)   
+        topology = common.load_models_from_args(parser, args)[0]
+
+        model_dir = download_dir / topology.subdirectory
+
+        flags = ['--name=' + model_name,
+                 '--output_dir=' + str(download_dir).replace('\\', '\\\\'), 
+                 '--cache_dir=' + str(cache_dir).replace('\\', '\\\\')]
+        if precision:
+            flags += ['--precisions=' + precision]
+        
+        downloader.main(flags)
+
+        self.name = model_name
+        self.model_dir = model_dir
+        self.download_dir = download_dir
+        self.precision = precision
+        self.precisions = topology.precisions
+        Topology.__doc__ = Topology.__doc__.format(topology.description, 
+                                                   topology.task_type, topology.license_url)
```

Comment:
This seems questionable. There's only one `Topology.__doc__`, so if you create multiple `Topology` objects, the doc will just be set to the documentation for the most recent one. I would suggest to just add a `help_text` attribute to the object.

Reply:
@likholat, please check once again something like this:
```
ssd = omz.Model('mobilenet-ssd')
alexnet = omz.Model('alexnet')
help(ssd)
```
what is output?

Reply:
Your code is works, but it always returns documentation for the first model, we couldn't do like:
```python
ssd = omz.Model('mobilenet-ssd')
detection=omz.Model('face-detection-0100')
help(detection)
```
It returns documentation for ```mobilenet-ssd```

Reply:
well, can it be fixed? In example by `self.__doc__` modification instead?

Reply:
FWIW, while it _is_ possible to reassign `self.__doc__`, it will not be used by Python. Currently, using `help` on an object will show you the documentation for the object's _class_. I found some discussion about adding documentation support for objects (https://bugs.python.org/issue35449), but it looks like nothing has materialized so far.

If you ask me, I think you should just have `description` and `task_type` properties in the `Topology` class. Given that you can't get help on object instances in Python in general, I don't think we need to make it work for our class.